### DISCUSSION TITLE
Dialog() Help [?]

### DISCUSSION THREAD

**Discussion by dbareis** [hluk/CopyQ#2024] [Q&A]
[Dialogs()](https://copyq.readthedocs.io/en/latest/scripting-api.html#dialog)  have a "?" button in the title bar, it doesn't do anything so I assume it ccan be made to do something.

It looks like it might be a two step help button but it's never helped... I'd like it to programatically build a URL and invoke it (single click).  Can this be done? Or can it be removed if truely useless?

I think using the "[QML helpRequested()](https://doc.qt.io/qt-6/qml-qtquick-controls2-dialog.html#helpRequested-signal)" will do something? Any example code?

![Screenshot @ 21_06_2022 , 10_57_41 AM, CopyQ +](https://user-images.githubusercontent.com/295230/174694809-ee284e74-c765-44ce-9932-11609bbeabb3.png)

**Comment by hluk** [comment]
That's some Windows feature that allows you to then click on a UI element to get some additional help. I do not think anything in CopyQ really uses it - additional help is usually shown as a tooltip when hovering over a UI element.

**Comment by dbareis** [comment]
It's a QT (not Windows) feature. If it's not used can it be turned off with the appropriate QT flag?

How would I setup tooltip help in dialog()?  Is it possible to add buttons to a dialog?

**Reply by hluk** [reply]
> It's a QT (not Windows) feature.

I haven't seen this on other systems than Windows, but I also remember seeing this a long time ago when developing for Windows in a different GUI framework.

> If it's not used can it be turned off with the appropriate QT flag?

I could not find how to do that (https://doc.qt.io/qt-5/qwhatsthis.html#details). I wonder if it shows only when some UI elements provides this help text - the default OK/Cancel buttons may have it.

> How would I setup tooltip help in dialog()? Is it possible to add buttons to a dialog?

This is not supported with `dialog()`.
